src/main/java/org/jitsi/videobridge/cc/BitrateController.java

```diff
@@ -599,20 +587,29 @@ public void constraintsChanged()
      * {@link ConferenceSpeechActivity}.
      * @param bweBps the current bandwidth estimation (in bps)
      */
-    private synchronized void update(List<AbstractEndpoint> conferenceEndpoints, long bweBps)
+    private synchronized void update()
     {
-        lastEndpointOrdering = conferenceEndpoints;
-        lastBwe = bweBps;
+        long nowMs = System.currentTimeMillis();
+
+        long bweBps = getAvailableBandwidth(nowMs);
```

Comment:
i don't see the advantage of having this method always re-get everything rather than having the other methods (`endpointOrderingChanged`, etc.)--which know exactly what changed--pass them into this method, so this method can focus just on dealing with the current state and not worrying about acquiring the current state. is there something i'm missing?

Reply:
but the `endpointOrderingChanged` doesn't know anything about bandwidth, nor `setMaxRxFrameHeightPx`. the last bandwidth estimation is part of the internal state that `update()` operates on and it decides to apply a hack to that bandwidth estimation that we may decide to remove in the future. i think the name `getAvailableBandwidth` is not representative but, conceptually, that method _is_ part of the update step

Reply:
that's true, but the thinking was that, for example `endpointOrderingChanged` gets the new endpoint ordering and knows to grab the existing/cached value for bandwidth, and `bandwidthChanged` would know to pass in the new bandwidth but use the cached endpoint ordering.  imagine calculating either of those things is costly--this method has to acquire both of them every time, since it doesn't know what changed.  having the other methods do that work ensures we only re-get the thing that actually changed.  that was the thinking, at least (as well as giving a more specific API for users to call--explicitly stating endpointOrderingChanged, etc. but we can achieve that with either of these schemes)

Reply:
i think i feel most strongly about `update` not being public (which it isn't) so i'm ok with this, i guess i just didn't see the need to change it from how it already was, as this way doesn't seem like an improvement.